state.go

```diff
@@ -129,8 +137,19 @@ type StateRunner interface {
 	// Obtain runner's state
 	GetState() State
 	// TODO generic state run action
+	RunState(ctx *StateContext) (State, bool)
```

Comment:
Kind of a helper that should have been here from the start. So instead of doing `StateRunner.GetState().Handle(ctx, ctrl)` we just call `StateRunner.RunState(ctx)`. Maybe it should take `Controller` as parameter too.